src/zenml/pipelines/base_pipeline.py

```diff
@@ -919,6 +853,30 @@ def connect(**steps: BaseStep) -> None:
 
         return connect
 
+    def _compile(
+        self, config_path: Optional[str] = None, **run_configuration_args: Any
+    ) -> Tuple["PipelineDeployment", "PipelineSpec"]:
+        # Activating the built-in integrations to load all materializers
+        from zenml.integrations.registry import integration_registry
+
+        integration_registry.activate_integrations()
```

Comment:
why is this needed here? feels a bit out of place

Reply:
This is needed so that all the materializers from integrations get registered in the registry (as the compilation below also verifies that materializers exist for each step output)